### ISSUE TITLE
Unable to run a transform on a value from reference

### ISSUE THREAD

**Issue by The-Code-Monkey** [amzn/style-dictionary#863]
So im doing this 

```
"primary": {
   "value": "{color.common.brand.primary.value}",
   "makeShades": true
}
```

The makeShades bool is a custom prop for my generate function to match and then do a transform on, it works fine when i put an actual value in the value prop, but if you put a reference like above it just completely skips all the transforms, how do i get it to pull the reference for that value first then run the transforms on the color??

This is kindof a strange situation but I cant work out why it wouldn't be setup to do references first then transforms.

**Comment by nahiyankhan** [comment]
In your custom transform, can you try with "transitive: true"? Reference: https://amzn.github.io/style-dictionary/#/transforms?id=transitive-transforms

I just recently made a color adjustment transform too (for pressed / disabled / hovers) and scratched my head over this. 😅

**Comment by The-Code-Monkey** [comment]
I will try tomorrow cheers

**Comment by Sebazzz** [comment]
I ran into this issue as well. `transitive` transforms won't work because that runs after resolving.

This appears to be hardcoded behaviour in style-dictionary currently:
https://github.com/style-dictionary/style-dictionary/blob/b13b9c7bd0b10a65d8995f1702037198dcd1fb03/lib/transform/token.js#L98-L120

**Comment by Sebazzz** [comment]
Come to think about it, this is in two places even, because these transforms then get invoked from `map.js`, which also has this (duplicate) filtering:

https://github.com/style-dictionary/style-dictionary/blob/b13b9c7bd0b10a65d8995f1702037198dcd1fb03/lib/transform/map.js#L67-L70

So it is not even possible to work around by using `transitive` or a different transformer type.

**Comment by jorenbroekema** [comment]
Yeah this is indeed hardcoded behavior in the fundaments of how tokens are resolved and transformed, I'm planning to completely refactor or rewrite it in v6 which will fix this issue. Not an easy task though
